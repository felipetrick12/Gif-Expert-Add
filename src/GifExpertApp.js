import React, {useState} from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";



export const GifExpertApp = () => {
//   const categories = ["One Samurai", "Dragon Ball", "Naruto"]; //es un arreglo que en react no puede ser mutado por que da error, para eso esta el hook de useState
//   const handleApp=()=> { //useState se utliza por que el componenente va cambiar de estado, setComponent para enviar la propiedad que va ser el cambio de estado del elemento
//         setCategories([...categories,'Yakuza']); //para agregar un nuevo elemento al array
        
//   }

  const [categories, setCategories] = useState(["Rick And Morty" ]);

 
  return (
    <>
      <h3>GifExpertAdd</h3>
        <AddCategory setCategories ={setCategories}/>
     
      <hr />
      <ol>
      {
            categories.map( category => (
            
                <GifGrid 
                key={category}
                category={category}
                 /> 
            ))
        }
      </ol>
    </>
  );
};
