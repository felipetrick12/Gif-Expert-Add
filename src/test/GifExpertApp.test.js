import React from 'react';
import {shallow} from 'enzyme'
import {GifExpertApp} from '../GifExpertApp'

describe('Probar Componenete GifExperAdd', () => {

    test('El componente tiene que pasar ', () => {
        const categories= ['naruto', 'Rick and morty'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
});
