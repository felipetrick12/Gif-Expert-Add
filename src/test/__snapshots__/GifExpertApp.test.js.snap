// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar Componenete GifExperAdd El componente tiene que pasar  1`] = `
<Fragment>
  <h3>
    GifExpertAdd
  </h3>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="naruto"
      key="naruto"
    />
    <GifGrid
      category="Rick and morty"
      key="Rick and morty"
    />
  </ol>
</Fragment>
`;
